### PR TITLE
feat: Add `File.read()` method

### PR DESCRIPTION
**Pull request by ArneGudermann**
This PR add a read Method for the `File` module so you can easy read files  from the Cloudstore.

### CODE REVIEW

File: src/viur/core/modules/file.py [hunk 1]
```diff
@@ -443,6 +444,34 @@ def write(self, filename: str, content: Any, mimetype: str = "text/plain", width
...
+            else:
+                path = f"""{skel["dlkey"]}/source/{skel["name"]}"""
+
+        blob = bucket.blob(path)
+        try:
+            data = blob.download_as_bytes()
+            return io.BytesIO(data)
+        except Exception as e:
+            logging.error(f"Could not read Bolb {e=}")
+            raise errors.NotFound()
```
Comment (phorward) [inline src/viur/core/modules/file.py:473]: IMHO this function is only used internally, should it really raise `NotFound`?
```suggestion
        data = blob.download_as_bytes()
        return io.BytesIO(data)
```

**Review by phorward — CHANGES_REQUESTED** [review]

File: src/viur/core/modules/file.py (hunk 1 above)
Comment (phorward) [reply src/viur/core/modules/file.py]: @ArneGudermann what about this proposal?

Comment (ArneGudermann) [reply src/viur/core/modules/file.py]: I missed it. Now ok ?

**Review by phorward — DISMISSED** [review]

File: src/viur/core/modules/file.py [hunk 2]
```diff
@@ -443,6 +444,31 @@ def write(self, filename: str, content: Any, mimetype: str = "text/plain", width
...
+        :param key: Key of the LeafSkel that contains the "dlkey" and the "name".
+        :param path: The path of the file in the Cloudstorage Bucket.
+
+        :return: Returns the file as a BytesIO buffer and the content-type
+        """
+        if not key and not path:
+            raise ValueError("Please provide a key or a path")
+        if key:
+            skel = self.viewSkel("leaf")
+            if not skel.fromDB(db.keyHelper(key, self.leafSkelCls().kindName)):
```
Comment (sveneberth) [inline src/viur/core/modules/file.py:463]:
```suggestion
            if not skel.fromDB(db.keyHelper(key, skel.kindName)):
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by phorward — APPROVED** [review]

File: src/viur/core/modules/file.py [hunk 3]
```diff
@@ -443,6 +444,31 @@ def write(self, filename: str, content: Any, mimetype: str = "text/plain", width
 
         return skel.toDB()
 
+    def read(self, key: db.Key | int | str | None = None, path: str | None = None) -> Tuple[io.BytesIO, str]:
+        """
+        Read a file from the Cloudstorage.
```
Comment (sveneberth) [inline src/viur/core/modules/file.py:449]:
```suggestion
        Read a file from the Cloud Storage.
```

File: src/viur/core/modules/file.py [hunk 4]
```diff
@@ -443,6 +444,31 @@ def write(self, filename: str, content: Any, mimetype: str = "text/plain", width
...
+
+        If a key and a path are provided, the key is preferred.
+        This means that the entry in the db is searched first and if this is not found, the path is used.
+
+        :param key: Key of the LeafSkel that contains the "dlkey" and the "name".
+        :param path: The path of the file in the Cloudstorage Bucket.
```
Comment (sveneberth) [inline src/viur/core/modules/file.py:455]:
```suggestion
        :param path: The path of the file in the Cloud Storage Bucket.
```

**Review by sveneberth — CHANGES_REQUESTED** [review]
Please correct the [product name](https://cloud.google.com/storage). Otherwise people can think we mean a different service.

**Review by sveneberth — APPROVED** [review]

**Pull request merged**